source/locales.tex

```diff
@@ -4632,3 +4651,719 @@
+other functions which can modify the environment\iref{support.runtime}.
+\end{itemdescr}
+
+\indexlibrarymember{environment_is}{text_encoding}%
+\begin{itemdecl}
+template<id i>
+static bool environment_is();
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\mandates
+\tcode{CHAR_BIT == 8} is \tcode{true}.
+
+\pnum
+\returns
+\tcode{environment() == i}.
+\end{itemdescr}
+
+\indexlibrarymember{\exposid{comp-name}}{text_encoding}%
+\begin{itemdecl}
+static constexpr bool @\exposid{comp-name}@(string_view a, string_view b);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{true} if the two strings \tcode{a} and \tcode{b}
+encoded in the ordinary literal encoding
+are equal, ignoring, from left-to-right,
+\begin{itemize}
+\item
+all elements that are not digits or letters\iref{character.seq.general},
+\item
+character case, and
+\item
+any sequence of one or more \tcode{0} characters
+not immediately preceded by a numeric prefix, where
+a numeric prefix is a sequence consisting of
+a digit in the range \crange{1}{9}
+optionally followed by one or more elements which are not digits or letters,
+\end{itemize}
+and \tcode{false} otherwise.
+
+\begin{note}
+This comparison is identical to
+the ``Charset Alias Matching'' algorithm
+described in the Unicode Technical Standard 22.
+\end{note}
+
+\begin{example}
+\begin{codeblock}
+assert(@\exposid{comp-name}@("UTF-8", "utf8") == true);
+assert(@\exposid{comp-name}@("u.t.f-008", "utf8") == true);
+assert(@\exposid{comp-name}@("ut8", "utf8") == false);
+assert(@\exposid{comp-name}@("utf-80", "utf8") == false);
+\end{codeblock}
+\end{example}
+\end{itemdescr}
+
+\rSec3[text.encoding.aliases]{Class text_encoding::aliases_view}
+
+\indexlibrarymember{aliases_view}{text_encoding}%
+\indexlibrarymember{begin}{text_encoding::aliases_view}%
+\indexlibrarymember{end}{text_encoding::aliases_view}%
+\begin{itemdecl}
+struct text_encoding::aliases_view : ranges::view_interface<text_encoding::aliases_view> {
+  constexpr @\textit{implementation-defined}@ begin() const;
+  constexpr @\textit{implementation-defined}@ end() const;
+};
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\tcode{text_encoding::aliases_view} models
+\libconcept{copyable},
+\tcode{ranges::\libconcept{view}},
+\tcode{ranges::\libconcept{random_access_range}}, and
+\tcode{ranges::\libconcept{borrowed_range}}.
+
+\pnum
+Both
+\tcode{ranges::range_value_t<text_encoding::aliases_view>} and
+\tcode{ranges::range_reference_t<text_encoding::aliases_view>}
+denote \tcode{const char*}.
+\end{itemdescr}
+
+\rSec3[text.encoding.comp]{Comparison functions}
+
+\indexlibrarymember{operator==}{text_encoding}%
+\begin{itemdecl}
+friend constexpr bool operator==(const text_encoding & a, const text_encoding & b) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+If \tcode{a.\exposid{mib_} == id::other \&\& b.\exposid{mib_} == id::other}
+is \tcode{true},
+then \tcode{\exposid{comp-name}(a.\exposid{name_}, b.\exposid{name_})}.
```

Comment:
super-ugly line break.  Please force a break after a.name_, (including the comma).